test/models/character_test.exs

```diff
@@ -0,0 +1,149 @@
+defmodule Yggdrasil.CharacterTest do
+  use Yggdrasil.ModelCase
+
+  alias Yggdrasil.User
+  alias Yggdrasil.Game
+  alias Yggdrasil.Character
+
+  @min_len 3
+
+  @user1   %{username: "tester",
+             password: "test123", 
+             password_confirmation: "test123" }
+
+  @user2   %{username: "tester2",
+             password: "test123", 
+             password_confirmation: "test123" }
+
+  @game1   %{name: "abcdef", description: "abcdefghi"}
+  @game2   %{name: "ghijkl", description: "abcdefghi"}
+
+  @character %{name: "foobar", user_id: nil, game_id: nil}
+  @invalid_character %{name: "ab", user_id: nil, game_id: nil}
+
+  @invalid_length_msg "should be at least %{count} character(s)"
+  @missing_assoc_msg "does not exist"
+  @unique_msg "has already been taken"
+
+  defp do_setup do
```

Comment:
There is a setup function that can be used that runs before each test, the reason I avoided it at the moment is because it's defined in the ModelCase and starts the db transaction which is imported via the using at the top.

I have few ideas around this that I need to review, but maybe we can clean thus up a bit.